File: infra/path_hacks/test/common_test.py
# Copyright 2015 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

import os
import unittest

import infra.path_hacks.common

class CommonPathTest(unittest.TestCase):
  def test_check_valid_common_path(self):
    # This supposes that all proper sibling directories have been checked out.
    common_path = infra.path_hacks.common._build_scripts_common
    self.assertTrue(os.path.isdir(common_path),
                    msg='Path %s cannot be found, make sure you have checked'
                        ' out this repository with gclient.' % common_path)

File: go/src/infra/libs/skylab/inventory/autotest/labels/common_test.go
// Copyright 2019 The Chromium OS Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package labels

import (
	"reflect"
	"testing"
)

func TestRemoveLabel(t *testing.T) {
	t.Parallel()
	got := []string{"ayanami", "laffey", "javelin"}
	got = removeLabel(got, 1)
	want := []string{"ayanami", "javelin"}
	if !reflect.DeepEqual(got, want) {
		input := []string{"ayanami", "laffey", "javelin"}
		t.Errorf("removeLabel(%#v, 1) = %#v; want %#v", input, got, want)
	}
}

func TestRemoveLabelOnLastItem(t *testing.T) {
	t.Parallel()
	got := []string{"ayanami", "laffey", "javelin"}
	got = removeLabel(got, 2)
	want := []string{"ayanami", "laffey"}
	if !reflect.DeepEqual(got, want) {
		input := []string{"ayanami", "laffey", "javelin"}
		t.Errorf("removeLabel(%#v, 2) = %#v; want %#v", input, got, want)
	}
}
